### DISCUSSION TITLE
Checking before upgrading

### DISCUSSION THREAD

**Discussion by hans0104** [dingo35/SmartEVSE-3.5#163] [General]
First I want to thank you Dingo for all your work making the smartevse more and more beautiful ;)
I'm still on the original software from Steegen. 
Main reason is that is works and I don't want to break my charger because I depend on it for charging my Tesla model y.
Questions:
- Can I just upgrade to 3.7.2 and at least have the same functionality as the original firmware?
- Is there a way to see what version I'm currently on, if for some reason I have to go back?
- Is it possible to keep the sensor box wired in stead of on wifi with 3.7.2? I already have it wired and I want to keep it that way :)

**Comment by dingo35** [comment]
It should all be backwards compatible, including the modbus connection between Sensorbox and SmartEVSE; if you dont need the extended functions of the Sensorbox you shouldnt even need to update the Sensorbox firmware.

**Reply by hans0104** [reply]
Just one additional question: 
If I upgrade the SensorBox with wifi, is the SmartEVSE still using the wired connection to measure the current?
That was actually my worry because I depend on the max charge current limiting.

**Reply by dingo35** [reply]
It should still work, but if I remember correctly there was a user reporting that after the upgrade his fix modbus connection didnt work anymore.
I couldnt reproduce it and didnt hear from it afterwards, so it might have been a "problem between keyboard and chair" ....

**Comment by dingo35** [comment]
BY THE WAY make sure you have bought the SmartEVSE v3, v2 or older is a whole other story!

**Reply by hans0104** [reply]
Thanks for your quick reply.
Yes, I have the v3 controller. 
Bought everything in januri of this year.
I will give it a try next week.
